### DISCUSSION TITLE
AqualinkD Web Page Controls Not Showing and Log Errors

### DISCUSSION THREAD

**Discussion by CodeNoMore** [aqualinkd/AqualinkD#369] [Q&A] [answered]
My installation of AqualinkD on a PDA will turn pool pump, lights on and off and read temps and SWG but the spa control is missing and the freeze protection control does not show up either. Seems like the other normal controls do show up. There are some errors and strange outputs in the log that are not clear to me. The swg cell has been removed for the winter but the control box is still connected so not sure if that is causing the specific swg unknown config error. Definitely could use some help to sort out these issues. Attached log from AqualinkD Manager.
Thanks.
[aqualinkdlog.txt](https://github.com/user-attachments/files/18044473/aqualinkdlog.txt)

**Comment by CodeNoMore** [comment]
Ok I was able to solve a few issues with a reboot and clearing the cache. So now the web controls are showing up but the spa on/off does not work and in AqualinkD Manager log I am getting these errors after a restart.
Error: RS Serial: Didn't find valid blocking serial port file descripter
Error:   AqualinkD: Unknown config parameter 'swg = Salt Water Generator'     (maybe because the cell is remove for winter??)
Error: PDA:       PDA Init :- wait for next menu failed
Error: PDA:       PDA Device programmer couldn't select menu item 'SYSTEM SETUP' menu 4
Error: PDA:       PDA Device programmer didn't find a requested menu 15, current 4
Error: PDA:       PDA Init :- Error getting freeze setpoints
Error: PDA:       PDA Device programmer couldn't find menu item 'Spa Mode      ' in menu 19 index -1
Error: PDA:       PDA Device On/Off, device 'Spa Mode' not found
Warning:NetService:WS: Didn't find device that matched URI 'Freeze_Protect/set'

Maybe these are related to some configuration issue.
Thanks.

**Reply by CodeNoMore** [reply]
The "Spa Mode" issue was resolved to a name change to just "Spa" which was an oversight from looking at the wrong menu.
These issues still remain.

Error: RS Serial: Didn't find valid blocking serial port file descripter
Error: AqualinkD: Unknown config parameter 'swg = Salt Water Generator' (maybe because the cell is remove for winter??)
Error: PDA: PDA Init :- wait for next menu failed
Error: PDA: PDA Device programmer couldn't select menu item 'SYSTEM SETUP' menu 4
Error: PDA: PDA Device programmer didn't find a requested menu 15, current 4
Error: PDA: PDA Init :- Error getting freeze setpoints
Warning:NetService:WS: Didn't find device that matched URI 'Freeze_Protect/set'

If anyone has some insights that would be helpful.
Thanks.

**Comment by sfeakes** [comment]
It looks like you may have some serial problems.  Check the USB2RS485 adapter.  It also looks like you are stoping and starting AqualinkD too quickly and not giving it time to connect and get all information from control panel.  Start it an wait a minute or two before doing anything, just watch the logs. 

Below is some more things from your logs.
```
Info:  Error:   AqualinkD: Unknown config parameter 'swg = Salt Water Generator'
```
Above had nothing to do with cell not being removed.  you have `swg = Salt Water Generator` in aqualinkd.conf and that's not a valid config entry.  I'm not sure what you are trying to do with that line, but delete it. (AqualinkD is simply ignoring that line and giving you a warning), so that's not hurting anything.

```
Warning:NetService:WS: Didn't find device that matched URI 'Freeze_Protect/set'
```
That looks like you are trying to hit the FreezeProtect button in the UI. So it must be showing up for you to be able to select it.  BUT, freeze protect can't be turned on / off, only change the turn on temperature.  Give it a long press for the temp UI to show up.

```
Error: PDA: PDA Init :- wait for next menu failed
Error: PDA: PDA Device programmer couldn't select menu item 'SYSTEM SETUP' menu 4
Error: PDA: PDA Device programmer didn't find a requested menu 15, current 4
Error: PDA: PDA Init :- Error getting freeze setpoints
Error: PDA: PDA Device programmer couldn't find menu item 'Spa Mode ' in menu 19 index -1
Error: PDA: PDA Device On/Off, device 'Spa Mode' not found
```

These are very serious errors, AqualinkD basically can't get to any menu items, so nothing will really work.

Please turn on debug logging let AqualinkD start and leave it for a minute, then post the complete log.

**Reply by CodeNoMore** [reply]
I deleted the SWG useless line and the warning did go away.
I was able to get the Freeze Protection web control to show up by forcing it to get the settings in the aqualinkd.conf file.
With the short click I was doing it says the set temp is -999 but when you hold it the slider control shows the correct value.
Freeze Protection on the web page always says it is off even when it is on.

The RS485 adapter is the same one I read in a different discussion thread on this forum that you used for your development and I did run the test in your Wiki and it gave the results as you stated so it does not seem on the surface that there is an issue there.


I have attached a log file from the AqualinkD Manager which was run for a minute but it seems to be limited to 500 lines.
I don't know if that will be useful but I enabled the logging capability and set to debug in the aqualinkd.conf but no log was created in the var/log directory on an aqualinkd restart.
What could be the issue there with no log as the permissions seem ok on the log directory?
Thank you for your assistance.

[AqualinkDManagerfile.log](https://github.com/user-attachments/files/18049617/AqualinkDManagerfile.log)

**Reply by sfeakes** [reply]
I need the log from when AqualinkD first starts up to debug this.  There is no limit on the log file size, if you are using aqmanager to download, then increase the lines to sat 5000.

![Screenshot 2024-12-07 at 1 29 32 PM](https://github.com/user-attachments/assets/0fa4eac9-df48-444b-bd29-04557918dfb2)
>
Also since we need this from when AqualinkD starts, you are best setting `log_level=DEBUG` in `/etc/aqualinkd.conf` rather that using aqmanager to set the debug mask after aqualinkd has started.

**Reply by CodeNoMore** [reply]
I think I got what you need if not I will try again.
Here is the debug log from aqmanager on a restart with the level set to debug in the conf file and 5000 lines.
Thanks.
[aqualinkd.log](https://github.com/user-attachments/files/18050409/aqualinkd.log)

**Reply by sfeakes** [reply, marked as answer]
From that log, it looks like everything is working.  You did stop and re-start AqualinkD though (I'm not sure why). So maybe you are not giving it long enough to start up and find all the data it needs. Remember PDA panels are very slow. And it will also take AqualinkD a long time to do anything with a PDA panel compared to RS.  One thing I did notice from your different logs though, is I think you need to slow AqualinkD down to work better with the PDA panel.
Change the frame delay from 0 to 4 in aqualinkd.conf, that should help.
```
rs485_frame_delay = 4
```

**Reply by CodeNoMore** [reply]
It seems like it has settled out and is working fine now so the slowness may be a factor. Changed the delay as you suggested.
Thanks for all your help and some great code.
